architectures/firmware/sof-zephyr/mpp_layer/async_messaging.rst

```diff
@@ -0,0 +1,40 @@
+.. _async_msg:
+
+Asynchronous Messaging Service
+##############################
+
+Asynchronous Messaging Service (AMS) is designed to exchange sporadic /
+asynchronous events between firmware components, for instance about key phrase
+detection. It is an optional component that can be selected in firmware build
+configuration. The service exposes external interface to host and API accessible
+from firmware components.
+
+Asynchronous messages are one-way from producer to all consumers, which
+allows to:
+
+  - direct asynchronous communication between components (1:1)
+  - sending one asynchronous message to many components (1:N)
+  - producing asynchronous messages by many modules and 1 is receiving (M:1)
+  - producing asynchronous messages by many modules and many is receiving (M:N)
```

Comment:
> I assume base FW is a supported sender/target ? May need to spell this out.

@lgirdwood yhe mechanism is designed for sending Async messages between components, some base FW management components can register and produce events if needed as well, but I would say that base FW is a set of components and not target/sender per se.

> and is host a target too ?

Currently no. Host notifications are send via IPC notification mechanism. Generally the AMS design can be extended for Host Async Messaging Gateway to support that, but currently it is not in scope.

Reply:
ok, it would be good to use AMS for host sending too i.e. one API for messaging and could be used today for Keyword notification. We could have this as a TODO though.